Lib/fontTools/misc/plistlib.py

```diff
@@ -52,19 +65,22 @@
 )
 
 
-def _date_from_string(s):
+def _date_from_string(s: str) -> datetime:
     order = ("year", "month", "day", "hour", "minute", "second")
-    gd = _date_parser.match(s).groupdict()
+    m = _date_parser.match(s)
+    if m is None:
+        raise ValueError(f"Expected ISO 8601 date string, but got '{s:r}'.")
+    gd = m.groupdict()
     lst = []
     for key in order:
         val = gd[key]
         if val is None:
             break
         lst.append(int(val))
-    return datetime(*lst)
+    return datetime(*lst)  # type: ignore
```

Comment:
why "ignore" here?

Reply:
Because mypy mubles something about expecting `tzdata` somewhere, which is the last argument we leave out.